torchgeo/datasets/satlas.py

```diff
@@ -0,0 +1,770 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SatlasPretrain dataset."""
+
+import os
+from collections.abc import Callable, Iterable
+from typing import ClassVar, TypedDict
+
+import numpy as np
+import pandas as pd
+import torch
+from einops import rearrange
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, check_integrity, extract_archive, which
+
+
+class _Task(TypedDict, total=False):
+    BackgroundInvalid: bool
+    categories: list[str]
+    colors: list[list[int]]
+    type: str
+
+
+# https://github.com/allenai/satlas/blob/main/satlas/model/dataset.py
+TASKS: dict[str, _Task] = {
+    'polyline_bin_segment': {
```

Comment:
Do you want to explicitly have these 400 lines of code here or maybe in a separate file that is part of the download? From what I can tell `TASKS` is only used in the plot function for the colors and could be loaded instead?

Reply:
I think it's fine, it's all copy-n-pasted anyway. Using downloadable code is bad without checksumming, although JSON might be okay. Could even shrink it if needed since we don't use all attributes.